tests/test-grammar-integration.cpp

```diff
@@ -841,6 +835,29 @@ static void test_json_schema() {
         }
     );
 
+    test_schema(
+        "additional properties can't override other properties",
+        R"""({
+            "properties": {
+                "a": {"type": "integer"},
+                "b": {"type": "integer"}
+            },
+            "additionalProperties": true
+        })""",
+        // Passing strings
+        {
+            "{\"a\": 42}",
```

Comment:
Nit: Would it be worth switch these to `R"""(`-type strings?

Reply:
Replaced most of them

Reply:
Replaced most of them